apps/src/applab/applab.js

```diff
@@ -1012,6 +1013,7 @@ Applab.reset = function () {
     designMode.resetPropertyTab();
   }
 
+  getStore().dispatch(disconnectMaker());
   if (makerBoard) {
     makerBoard.destroy();
     makerBoard = null;
```

Comment:
Should disconnectMaker() be dispatched just after the board is destroyed? It seems like there could theoretically be a race condition if the reset button is pressed just before the board receives a connection error:
* thread 1: `getStore().dispatch(disconnectMaker());` is executed, hiding the "connection pending" overlay
* thread 2: `board.connect()` fails
* thread 2: `dispatch(reportConnectionError())` is executed, showing the "connection error" overlay
* thread 1: `makerBoard.destroy()`
presumably `board.destroy()` cancels whatever is pending in `board.connect()`.